frame/elections-phragmen/src/lib.rs

```diff
@@ -334,18 +353,32 @@ decl_module! {
 		///
 		/// A defunct voter is defined to be:
 		///   - a voter whose current submitted votes are all invalid. i.e. all of them are no
-		///     longer a candidate nor an active member.
+		///     longer a candidate nor an active member or a runner-up.
+		///
+		///
+		/// The origin must provide teh number of current candidates for the purpose of accurate
+		/// weight calculation.
 		///
 		/// # <weight>
 		/// #### State
 		/// Reads: O(NLogM) given M current candidates and N votes for `target`.
 		/// Writes: O(1)
 		/// # </weight>
 		#[weight = 1_000_000_000]
-		fn report_defunct_voter(origin, target: <T::Lookup as StaticLookup>::Source) {
+		fn report_defunct_voter(
+			origin,
+			target: <T::Lookup as StaticLookup>::Source,
+			#[compact] candidate_count: u32,
+		) {
 			let reporter = ensure_signed(origin)?;
 			let target = T::Lookup::lookup(target)?;
 
+			let actual_count = <Candidates<T>>::decode_len().unwrap_or(0);
+			// TODO: refund based on actual is less than predicted.
+			ensure!(
+				actual_count as u32 == candidate_count,
```

Comment:
This should be a less than or equal to check, allowing a user to put a slightly higher number and avoid creating an invalid tx

Reply:
I did that at first, but then it caused some flaky tests and reverted it. Sure can do, it is not that big of a difference. Basically we tolerate it if you want to over charge yourself.

Reply:
Yeah, honestly better to overcharge a little than to return error and force the user to lose their whole fee!

Reply:
done